src/api-service/__app__/onefuzzlib/workers/scalesets.py

```diff
@@ -551,6 +545,9 @@ def delete_nodes(self, nodes: List[Node]) -> None:
             machine_ids,
         )
         delete_vmss_nodes(self.scaleset_id, machine_ids)
```

Comment:
So this begins an async deletion operation, and internally gets access to an `LROPoller`. We ignore that, and return `Option[Error]` In practice, it looks like we always return `None`.

Does the initial state of the poller contain data we should be examining? If deletion fails right away, how do we detect it?

Reply:
(or if deletion fails later, when do we detect it?)

Reply:
To answer your question about the LROPoller, the initial state of the poller isn't immediately useful.  The request itself fails if it's immediately bad.  I've already started investigating using the LROPoller.from_continuation_token to reconstruct the poller, but that's a relatively large change.

With regarding recovery from Azure failures, it will take some time, but we'll eventually recover.

* For reimaging, `cleanup_nodes` will recreate it, the heartbeat will eventually be too long expired, and we'll reimage it.
* For deletion, `cleanup_nodes` will recreate it and we'll reimage it like above, but now `sync_scaleset_size` will eventually trigger deletion of a node.

Reply:
Makes sense, and the deeper fix to the eventing complexity is out of scope here.